public/js/components/SourcesTree.js

```diff
@@ -1,18 +1,16 @@
 const React = require("react");
 const { DOM: dom, PropTypes } = React;
+const classnames = require("classnames");
+const ImPropTypes = require("react-immutable-proptypes");
+const { Set } = require("immutable");
 
 const {
   nodeHasChildren, createParentMap, addToTree,
   collapseTree, createTree
 } = require("../utils/sources-tree.js");
-
-const classnames = require("classnames");
-const ImPropTypes = require("react-immutable-proptypes");
-const { Set } = require("immutable");
-const debounce = require("lodash/debounce");
-
 const ManagedTree = React.createFactory(require("./utils/ManagedTree"));
 const Svg = require("./utils/Svg");
+const { throttle } = require("../utils/utils");
```

Comment:
`Creates a throttled function that only invokes func at most once per every wait milliseconds` That's exactly what this does, if it doesn't then it's a bug and I'll fix it :) It's the same as throttling with async channels, etc, which is what I'm familiar with.